### PR TITLE
[relates #36 #190] Add basic e2e tests with cucumber.js

### PR DESCRIPTION
**Pull request by jludvice**

### CODE REVIEW

**Comment by jludvice** [comment]
Why is this jenkins test failing?
```
[redhat-ipaas-ui-pullreq] $ /bin/sh -xe /tmp/hudson663253643725566418.sh
+ yarn test:xvfb
/mnt/hudson_workspace/tools/jenkins.plugins.nodejs.tools.NodeJSInstallation/node_7.4.0/lib/node_modules/yarn/bin/yarn.js:61
      throw err;
      ^

Error: ENOENT: no such file or directory, open '/mnt/hudson_workspace/tools/jenkins.plugins.nodejs.tools.NodeJSInstallation/node_7.4.0/lib/node_modules/yarn/node_modules/end-of-stream/node_modules/once/once.js'
    at Object.fs.openSync (fs.js:558:18)
    at Object.fs.readFileSync (fs.js:468:33)
    at Object.Module._extensions..js (module.js:579:20)
    at Module.load (module.js:488:32)
    at tryModuleLoad (module.js:447:12)
    at Function.Module._load (module.js:439:3)
    at Module.require (module.js:498:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/mnt/hudson_workspace/tools/jenkins.plugins.nodejs.tools.NodeJSInstallation/node_7.4.0/lib/node_modules/yarn/node_modules/end-of-stream/index.js:1:74)
    at Module._compile (module.js:571:32)
Build step 'Execute shell' marked build as failure
```

**Comment by kahboom** [comment]
@jludvice - Jenkins hasn't been working in a while, but your circleCI tests should at least pass. Or at least one of the checks should.

**Comment by jludvice** [comment]
@kahboom circle ci is unhappy about my tslint - mostly `console.log` and `else if` statement.
Do I have to switch to your logging implementation?
I don't have any angular context in e2e tests and my `console.log` is actually printed into bash terminal, not into browser console.

```                
yarn lint v0.21.3
$ yarn ng lint 
yarn ng v0.21.3
$ "/home/jludvice/devel/redhat/ipaas/ipaas-ui/node_modules/.bin/ng" lint
e2e/app.po.ts[80, 5]: Calls to 'console.log' are not allowed.
e2e/app.po.ts[110, 5]: misplaced 'else'
e2e/common/common.steps.ts[43, 5]: Calls to 'console.log' are not allowed.
e2e/connections/list/list.po.ts[15, 5]: Calls to 'console.log' are not allowed.
e2e/connections/list/list.po.ts[21, 5]: Calls to 'console.log' are not allowed.
/home/jludvice/devel/redhat/ipaas/ipaas-ui/e2e/login/login.po.ts[32, 5]: Calls to 'console.log' are not allowed.
/home/jludvice/devel/redhat/ipaas/ipaas-ui/e2e/login/login.po.ts[34, 7]: Calls to 'console.log' are not allowed.
e2e/login/login.steps.ts[24, 5]: Calls to 'console.log' are not allowed.
e2e/narratives/08-first-pass.steps.ts[25, 5]: Calls to 'console.log' are not allowed.
e2e/narratives/08-first-pass.steps.ts[33, 5]: Calls to 'console.log' are not allowed.
Lint errors found in the listed files.
error Command failed with exit code 2.
error Command failed with exit code 1.
```

**Comment by kahboom** [comment]
@jludvice - Yeah you'd need to switch to our logger. @gashcrumb may be able to chime in here as I haven't even taken a look at it tbh.

**Comment by gashcrumb** [comment]
@jludvice you could disable linting for your own file if you like:

```
/* tslint:disable */
```

at the top.  Think you can also disable individual rules, but you'll need to look that up, I don't recall off the top of my head.

**Comment by jludvice** [comment]
Unfortunately it looks like tslint doesn't support configuration cascading yet :( 
https://github.com/palantir/tslint/issues/2302

I also tried to import logging from `src/app` with no luck.
I guess it doesn't like when it runs outside of browser where is no `window` defined.
```
[11:01:22] E/launcher - Error: ReferenceError: window is not defined
    at Function.ExtensionHelper.register (/home/jludvice/devel/redhat/ipaas/ipaas-ui/node_modules/typescript-logging/src/logging/extension/ExtensionHelper.ts:37:7)
    at Object.<anonymous> (/home/jludvice/devel/redhat/ipaas/ipaas-ui/node_modules/typescript-logging/src/logging/typescript-logging.ts:41:17)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/home/jludvice/devel/redhat/ipaas/ipaas-ui/src/app/logging.ts:1:1)
```

Will explore more options.

**Comment by jimmidyson** [comment]
retest this please

**Comment by jimmidyson** [comment]
retest this please

File: src/app/logging.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import { Category, CategoryLogger, CategoryServiceFactory, CategoryDefaultConfiguration, LogLevel } from 'typescript-logging';
 
 // TODO typescript-logging will support a console API to change this at runtime in the future, for now we'll use localStorage
-const storedLogLevel = localStorage.getItem('logLevel');
+const storedLogLevel = typeof localStorage !== 'undefined' && localStorage.getItem('logLevel');
```
Comment (jludvice) [inline src/app/logging.ts]: @gashcrumb are you ok with this change and `typescript-logging` version bump (#317)?

Comment (gashcrumb) [inline src/app/logging.ts]: @jludvice yeah, that's fine!

File: README.md [hunk 2]
```diff
@@ -12,7 +12,7 @@ Included in this stack are the following technologies:
 * Framework: [Angular 2](https://angular.io/)
 * Module Bundler: [Angular CLI](https://cli.angular.io)
 * Design Patterns: [PatternFly](https://www.patternfly.org/)
-* Testing: [Jasmine](http://jasmine.github.io/) (BDD Unit Test Framework), [Karma](https://karma-runner.github.io/1.0/index.html) (Unit Test Runner), [Protractor](http://www.protractortest.org/#/) (E2E Framework), [Istanbul](https://github.com/gotwarlost/istanbul) (Code Coverage)
+* Testing: [Cucumber.js](https://cucumber.io/) (BDD Unit Test Framework), [Karma](https://karma-runner.github.io/1.0/index.html) (Unit Test Runner), [Protractor](http://www.protractortest.org/#/) (E2E Framework), [Istanbul](https://github.com/gotwarlost/istanbul) (Code Coverage)
```
Comment (kahboom) [inline README.md]: Thanks for this.

File: e2e/app.po.ts [hunk 3]
```diff
@@ -1,11 +1,119 @@
-import { browser, element, by } from 'protractor';
+import { by, browser, element, ExpectedConditions } from 'protractor';
+import * as webdriver from 'selenium-webdriver';
+import { Promise as P } from 'es6-promise';
```
Comment (kahboom) [inline e2e/app.po.ts]: I appreciate meeting about this; it's totally okay for now, and not as many P's as I thought. 👍

File: e2e/login/login.feature [hunk 4]
```diff
@@ -0,0 +1,17 @@
+# Created by jludvice at 1.3.17
+@login
+Feature: First pass at login, homepage, connections
```
Comment (kahboom) [inline e2e/login/login.feature]: This user story and narrative approach using Cucumber is really nice and fits well with agile methodologies.

**Pull request merged**